File: 2015/day1.jl
data = open("./input/day1.txt") do file
    read(file, String)
end
length(data)

let floor = 0
    let pos  = 0

        for char in split(data, "")
            pos += 1
            if char == "("
                floor += 1
            elseif char == ")"
                floor -= 1
            end
            if floor == -1
                println(pos)
                break
            end
        end

        println(floor)
    end
end

File: 2025/day1.py
data = []
with open("input/day1.txt", "r") as f:
    for line in f:
        data.append((line[:1], int(line[1:].strip())))

pointer = 50
n_exact_zeros = 0
n_zero_crosses = 0

for direction, magnitude in data:
    if direction == "R":
        k0 = (100 - pointer) % 100
        pointer = (pointer + magnitude) % 100
    else:
        k0 = pointer % 100
        pointer = (pointer - magnitude) % 100

    if k0 == 0:
        k0 = 100
    if magnitude >= k0:
        n_zero_crosses += 1 + (magnitude - k0) // 100
    if pointer == 0:
        n_exact_zeros += 1

print(f"part1: {n_exact_zeros} part2: {n_zero_crosses}")
